### PR TITLE
doc/dev: update leads list

### PR DESCRIPTION
**Pull request by zdover23**
Update the list of component leads in
doc/dev/developer_guide/essentials.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/dev/developer_guide/essentials.rst [hunk 1]
```diff
@@ -13,20 +13,19 @@ following table shows all the leads and their nicks on `GitHub`_:
...
-========= ================ =============
+========= ================== =============
+Scope     Lead               GitHub nick
+========= ================== =============
+RADOS     Radoslaw Zarzynski rzarzynski 
+RGW       Casey Bodley       cbodley 
+RGW       Matt Benjamin      mattbenjamin
+RBD       Ilya Dryomov       dis 
+CephFS    Venky Shankar      vshankar
+Dashboard Nizamudeen A       nizamia109
```
Comment (nizamial09) [inline doc/dev/developer_guide/essentials.rst:24]: ;)
```suggestion
Dashboard Nizamudeen A       nizamial09 
```

Comment (zdover23) [reply doc/dev/developer_guide/essentials.rst:24]: What am I missing? I don't see a difference. I've been awake for a very long time, so I might be a little dumb right now.

Comment (nizamial09) [reply doc/dev/developer_guide/essentials.rst:24]: You put `1` instead of the alphabet `L` before `09`

Comment (nizamial09) [reply doc/dev/developer_guide/essentials.rst:24]: It looks so similar...

Comment (zdover23) [reply doc/dev/developer_guide/essentials.rst:24]: Aw man. Our orthography is almost as dumb as I was in this instance.

Thanks for pointing this out, @nizamial09.

Comment (nizamial09) [reply doc/dev/developer_guide/essentials.rst]: no problem!

**Review by anthonyeleven — APPROVED** [review]
Monitor not MON.  Not an acronym.  Or Mon.

**Review by cbodley — APPROVED** [review]

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/56588 - Squid backport
https://github.com/ceph/ceph/pull/56589 - Reef backport
https://github.com/ceph/ceph/pull/56590 - Quincy backport

File: doc/dev/developer_guide/essentials.rst [hunk 2]
```diff
@@ -13,20 +13,19 @@ following table shows all the leads and their nicks on `GitHub`_:
...
+========= ================== =============
+Scope     Lead               GitHub nick
+========= ================== =============
+RADOS     Radoslaw Zarzynski rzarzynski 
+RGW       Casey Bodley       cbodley 
+RGW       Matt Benjamin      mattbenjamin
+RBD       Ilya Dryomov       dis 
+CephFS    Venky Shankar      vshankar
+Dashboard Nizamudeen A       nizamial09 
+MON       Joao Luis          jecluis
```
Comment (batrick) [inline doc/dev/developer_guide/essentials.rst:25]: TMK Joao is no longer working on Ceph or maintaining the monitors anymore.

Comment (zdover23) [reply doc/dev/developer_guide/essentials.rst:25]: @batrick: Thanks, Patrick. I'll make a follow-up PR when I'm done with breakfast.

Comment (zdover23) [reply doc/dev/developer_guide/essentials.rst:25]: https://github.com/ceph/ceph/pull/56600 - This PR removes Joao from the list of leads.

**Pull request merged**